File: dongmin/week3/가까운 수.java
/*특이한 정렬과 비슷한 문제같아 보여 특이한 정렬을 응용해서 풀어봤는데
다른 사람의 풀이 방법을 보니 훨씬 간단하게 풀 수 있는 문제였다. 
먼저 array를 오름 차순 정렬한 후 각각 정수 n값 을 뺀 후 절대 값을 붙여
그 크기가 0에 더 가까운 수를 0번째 인덱스에 위치하게 한 뒤 0번째 인덱스 값을
return하면 쉽게 풀 수 있었다
*/

import java.util.Arrays;

class Solution {
    public int solution(int[] array, int n) {
        int answer = 0;
        double []result=new double[array.length];   //array 배열이 n보다 클 경우 소수점을 받기 위해 double형으로 선언
        int k=0;
        for (int i = 0; i < array.length; i++) {
            if(array[i] > n)       
                result[i]=array[i]-n+0.1;
            else
                result[i]=Math.abs(array[i]-n);     //값이 n보다 작으면 계산 후 절대값을 씌운다.
        }

        Arrays.sort(result);    //오름차순 정렬
        if(result[0]%1==0)      //n이 더 클 때
            array[0]=n-(int)result[0];
        else
            array[0]=(int)result[0]+n;
        answer=array[0];
        return array[0];
    }
}

File: yongkeun/3week/가까운수.js
function solution(array, n) {
  // 하나씩 방문하여 절대값을 비교함, 만약 절대값과 result 값이 같다면 작은 값을 result에 넣음
  // 만약 새로 들어온 절대값이 더 작다면 result을 바꿈
  let result = array[0]; //첫번째 값을 result에 넣고 시작
  for (let i = 1; i < array.length; i++) {
    if (Math.abs(result - n) == Math.abs(array[i] - n)) {
      //1번째 방의 절대값과 result가 같다면
      result = Math.min(result, array[i]); //두개중에 작은 값을 result에 넣음
    } else if (Math.abs(result - n) > Math.abs(array[i] - n)) {
      //같진 않았고, 이번 절대값이 더 작으면
      result = array[i]; //이번 인덱스의 값을 result로 바꿈
    }
  }
  return result;
}
